core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java

```diff
@@ -3383,6 +3383,16 @@ void checkCorrelatedMapSubQuery(boolean expand) {
     sql(sql).withDecorrelate(true).ok();
   }
 
+  /**
+   * Test case for <a href="https://issues.apache.org/jira/browse/CALCITE-5127">[CALCITE-5127]
+   * Error when executing query with subquery in select list that uses outer column of array
+   * type</a>.
+   */
+  @Test void testCorrelationWithProjection() {
```

Comment:
Possibly more descriptive names for the test case:
* `testNotMergeProjectionsWhenCorrelationPresent`
* `testProjectionWithCorrelationNoMerge`